### PR TITLE
Code optimizations reported by static code analysis (2020-11-07)

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

SUMMARY: Performance "Code optimizations reported by static code analysis"

#### Purpose of change

Cleanup code while increase performance and fix minor errors.

#### Describe the solution

See commit messages.

### CODE REVIEW

File: src/item_factory.cpp [hunk 1]
```diff
@@ -558,7 +558,7 @@ void Item_factory::finalize_post( itype &obj )
     }
 
     if( obj.comestible ) {
-        for( const std::pair<diseasetype_id, int> elem : obj.comestible->contamination ) {
+        for( const std::pair<diseasetype_id, int> &elem : obj.comestible->contamination ) {
```
Comment (Qrox) [inline src/item_factory.cpp:561]:
```suggestion
        for( const std::pair<const diseasetype_id, int> &elem : obj.comestible->contamination ) {
```

File: src/map.h [hunk 2]
```diff
@@ -1788,8 +1788,8 @@ class map
         void add_light_source( const tripoint &p, float luminance );
         // Handle just cardinal directions and 45 deg angles.
         void apply_directional_light( const tripoint &p, int direction, float luminance );
-        void apply_light_arc( const tripoint &p, units::angle, float luminance,
-                              units::angle wideangle = 30_degrees );
+        void apply_light_arc( const tripoint &p, const units::angle &dir, float luminance,
```
Comment (Qrox) [inline src/map.h:1791]:
```suggestion
        void apply_light_arc( const tripoint &p, const units::angle &angle, float luminance,
```

**Comment by akrieger** [comment]
Why the removal of the BufferSecurityCheck disable? Because of the warning about /sdl- and /GS- overriding eachother? That's a known issue apparently (https://developercommunity.visualstudio.com/content/problem/309307/d9025-overriding-sdl-with-gs.html) but I'm not sure why we should pay the perf hit for it.

**Comment by ZhilkinSerg** [comment]
Oh, thanks - that was actually a stray commit. It should've not been added to this branch.

**Pull request merged**